.github/workflows/continous-integration.yml

```diff
@@ -0,0 +1,123 @@
+# - PYPI_TOKEN: publishing token for tmbo account, needs to be maintainer of
+#               tmbo/questionary on pypi
+
+jobs:
+  quality:
+    name: Code Quality
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout git repository 🕝
+      uses: actions/checkout@v2
+
+    - name: Set up Python 3.7 🐍
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install poetry 🦄
+      uses: Gr1N/setup-poetry@v3
+
+    - name: Load Poetry Cached Libraries ⬇
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pypoetry
+        key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+        restore-keys: ${{ runner.os }}-poetry-
+
+    - name: Install dependencies 🖥
+      run: poetry install --no-interaction
+
+    - name: Lint Code 🎎
+      run: make lint
+
+    - name: Check Types 📚
+      run: make types
+
+    - name: Check Version Numbers 🕸
+      run: python3 scripts/validate_version.py
+
+  test:
+    name: Run Tests
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8]
+        promttoolkit: [3.*, 2.*]
+        exclude:
+          # tests on windows with prompt-toolkit 2.* do not complete
+          - os: windows-latest
+            promttoolkit: 2.*
+
+    steps:
+    - name: Checkout git repository 🕝
+      uses: actions/checkout@v2
+
+    - name: Set up Python 3.7 🐍
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install poetry 🦄
+      uses: Gr1N/setup-poetry@v3
+
+    - name: Load Poetry Cached Libraries ⬇
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pypoetry
+        key: ${{ runner.os }}-poetry-${{ hashFiles('**/poetry.lock') }}
+        restore-keys: ${{ runner.os }}-poetry-
+
+    - name: Install dependencies 🖥
+      run: |
+        poetry install --no-interaction
+        poetry run pip install prompt_toolkit==${{ matrix.promttoolkit }}
+
+    - name: Test Code 🔍
+      run: make test
+
+    - name: Send Coverage Report 📊
+      if: matrix.python-version == 3.6 && matrix.os != 'windows-latest'
+      env:
+        COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        COVERALLS_SERVICE_NAME: github
+      run: poetry run coveralls
+
+  deploy:
+    name: Deploy to PyPI
+    runs-on: ubuntu-latest
+
+    # deploy will only be run when there is a tag available
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    needs: [quality, test]  # only run after all other stages succeeded
+
+    steps:
+    - name: Checkout git repository 🕝
+      uses: actions/checkout@v2
+
+    - name: Set up Python 3.7 🐍
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Install poetry 🦄
+      uses: Gr1N/setup-poetry@v3
+
+    - name: Build ⚒️ Distributions
+      run: poetry build -u __token__ -p ${{ secrets.PYPI_TOKEN }}
```

Comment:
Do you think it might be nicer to have a new GitHub action that is run every time a new release is created (and not on each pull request): https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#release

This action would deploy the new release to PyPi

Reply:
well, we still want to make sure that tests pass before releasing. it wasn't possible to do that when github actions got released, not sure if they added something more recently to allow depending on other workflows

Reply:
I think if code is only ever added to `master` via a pull request, then it is guaranteed to have passed the tests. The flow would be something like this:

1. PR comes in.
2. Tests run.
3. If tests have passed then merge into master.
4. (No workflows are run now).
5. Create a release.
6. Run the deploy workflow to push to PyPi.

Reply:
That's a good point!